src/CRISP.BackendChallenge/Controllers/EmployeeController.cs

```diff
@@ -1,39 +1,193 @@
+    public IActionResult SearchEmployees(string? name, int? department)
+    {
+        _logger.LogDebug(":: Performing {MethodName}", nameof(SearchEmployees));
+
+        var query = _employeeRepository.Query();
+        
+        if (!string.IsNullOrWhiteSpace(name))
+            query = query.Where(x => x.Name == name);
+
+        if (department.HasValue)
+            query = query.Where(x => x.Department == (Department)department);
+
+        
+        query = query.Include(e => e.Logins);
+        
+        var result = query.Select(x => new EmployeeResponse
+        {
+            Id = x.Id,
+            Name = x.Name,
+            LoginDates = x.Logins.Select(l => l.LoginDate).ToList()
+        }).ToList();
+        
+        return Ok(result);
+    
+        
+    }
+    
+    
+    [HttpPut]
+    [Route("update/{id:int}")]
+    [TypeFilter(typeof(EmployeeModelStateFilter))]
+    public IActionResult UpdatedById(int id,[FromBody] EmployeeRequest employeeRequest) 
+    {
+        _logger.LogDebug(":: Performing {MethodName} with ID {Id}", nameof(UpdatedById), id);
+        
+        try
+        {
+            var logins = employeeRequest.LoginDates?.Select(loginDate => new Login {
+                EmployeeId = id,
+                LoginDate = loginDate
+            }).ToList();
+            
+            var employee = new Employee
+            {
+                Id = id,
+                Name = employeeRequest.Name!,
+                Department = (Department)employeeRequest.Department!,
+                StartDate = employeeRequest.StartDate,
+                EndDate = employeeRequest.EndDate,
+                Logins = logins!
+            };
+            
+            var updatedEmployee = _employeeRepository.Update(employee);
+
+            return Ok(new EmployeeResponse
             {
-                Id = x.Id,
-                Name = x.Name,
-                // TODO: Include the login date information...
-                LoginDates = null
+                Id = updatedEmployee.Id,
+                Name = updatedEmployee.Name,
+                LoginDates = updatedEmployee.Logins.Select(l => l.LoginDate).ToList()
             });
-        return Ok(result);
+        }
+        catch (KeyNotFoundException e)
+        {
+            _logger.LogDebug(":: Employee Not Found: {Id}, {EMessage}", id, e.Message);
+            return CreateNotFound(id);
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: Employee Not Found: {Id}, {EMessage}", id, e.Message);
+            return CreateNotFound(id);
+        }
+    }
+    
+
+    [HttpDelete]
+    [Route("delete/{id:int}")]
+    public IActionResult DeleteById(int id)
+    {
+        _logger.LogDebug(":: Performing {MethodName} with ID {Id}", nameof(DeleteById), id);
+
+        var employee = _employeeRepository.GetById(id);
+        
+        if (employee == null!) return CreateNotFound(id);
+        
+        _employeeRepository.Delete(employee);
```

Comment:
As pointed out before - each repository method should fully encapsulate a database command. In this instance, you are using two separate repository methods for a command.

In its current form, the `Delete` method is useless to your application on its own